support/sier-codec/src/object.rs

```diff
@@ -11,6 +11,15 @@ impl Object<'_> {
     pub fn schema(&self) -> &StructDef {
         self.schema
     }
+
+    pub fn serialize(&self) -> Vec<u8> {
+        let mut result = Vec::new();
+        result.extend(self.schema.id().iter().cloned());
+        for value in &self.values {
+            result.extend(value.serialize());
```

Comment:
Out of curiosity, why the use of `std::vec::Vec::extend` rather than `push`?

Reply:
Extend is basically push for when you want to push many things that are already in an iterator.

Also, extend can be more efficient when the size of the iterator you're extending with is known. Basically because it can allocate once if necessary, where push may allocate and move the contents multiple times.

Reply:
Extend is push when you want to push all the values in an iterator.

Also, extend can be more efficient if the size of the iterator is known. Then the Vector only has to allocate at most once, when with a sequence of pushes, it may have to reallocate many times.